packages/client/src/Components/PaperEdits/PaperCutsPlayer/ProgressBar.js

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable template-curly-spacing */
+/* eslint-disable react/prop-types */
+import React from 'react';
+
+// import './styles.css';
+
+class PaperCutsPlayerProgressBar extends React.PureComponent {
+  constructor(props) {
+    const {
+      width = 640,
+    } = props;
+
+    super(props);
+
+    this.width = width;
+    this.state = { progress: 0 };
+  }
+
+  componentDidUpdate() {
+    this.videoContext = this.props.videoContext;
+    this.setState({ duration: this.videoContext.duration });
+    this.updateProgress();
+  }
+
+  updateProgress = () => requestAnimationFrame(() => {
+    const currentTime = this.videoContext && this.videoContext.currentTime;
+    const progress = (currentTime / this.state.duration) * 100;
+
+    if (this.state.progress !== progress) this.setState({ progress });
+
+    requestAnimationFrame(this.updateProgress);
```

Comment:
if you do add some notes in the docs, some more details about the use of [`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) would also be good. In relation to optimising the `currentTime` updates etc..

Reply:
@melnyczuk  Added a draft with some links on `requestAnimation` frame